crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1512,7 +1512,23 @@ impl Room {
 
                 if !self.are_members_synced() {
                     self.sync_members().await?;
-                    // TODO query keys here?
+
+                    // Query keys in case we don't have them for newly synced members.
+                    let olm = self.client.olm_machine().await;
```

Comment:
Let's put this into a method, we're going to likely need to expose such a method publicly one of these days.